src/__tests__/user.spec.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable node/no-unpublished-require */
+const mongoose = require('mongoose');
+const supertest = require('supertest');
+const app = require('../app');
+const UserModel = require('../models/user');
+const connectDatabase = require('../db/connection');
+
+beforeAll(async () => {
+  connectDatabase();
+});
+
+afterAll(async () => {
+  await UserModel.deleteMany({});
+  await mongoose.disconnect();
+  await mongoose.connection.close();
+});
+
+describe('GET /api/user/:id', () => {
+  const invalidUserId = '62ee5c280262c62f56124dfd';
+
+  it('should return a found user', (done) => {
```

Comment:
Do you mean it should return an error instead of a found user?

Reply:
Nope. It is supposed to return a real user, but since it doesn't have a token, it also doesn't have access to that page. Token requirement could've been added later, after all required test were implemented and tested, maybe.

Reply:
The is not explanatory, it confused me. Also you can check if there is a user does the endpoint return the user. If there is no user which means no token then it should return an error.

Reply:
I thought of doing that, but couldn't really achieve it. I tested by commenting the token verification middleware; it worked; I thought that'd be enough. If adding a token, and making a "signed-in user", in a test case is possible, could you please send it to me; I can change the tests right away.